### PR TITLE
fix an error when changing several cluster node groups in parallel

### PR DESCRIPTION
**Pull request by morozovalekseywot**

### CODE REVIEW

File: vkcs/kubernetes/shared.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/util/errutil"
 )
 
+func clusterMutexKey(clusterUUID string) string {
```
Comment (paaanic) [inline vkcs/kubernetes/shared.go:15]: Looks like you use it only in one file for node group resource, so it would be better to place it there.

**Review by paaanic — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ description: |-
 - Fix panic on empty taint to vkcs_kubernetes_node_group resource
 - Fix order-induced changes in the plan for "databases" of the vkcs_db_user resource
 - Fix panic on VKCS Kubernetes API error in vkcs_kubernetes_clustertemplates data source
+- Fix an error when changing several cluster node groups in parallel
```
[the remark this answers is not in the document]
Comment (morozovalekseywot) [reply CHANGELOG.md]: done

File: vkcs/kubernetes/resource_node_group.go [hunk 3]
```diff
@@ -383,6 +383,13 @@ func resourceKubernetesNodeGroupUpdate(ctx context.Context, d *schema.ResourceDa
 			Delta: d.Get("node_count").(int) - s.NodeCount,
 		}
 
+		if !locked {
+			mutex.Lock(clusterMutexKey(clusterID))
+			// nolint:ineffassign
+			locked = true
```
[the remark this answers is not in the document]
Comment (morozovalekseywot) [reply vkcs/kubernetes/resource_node_group.go]: done

File: vkcs/kubernetes/shared.go (hunk 1 above)
Comment (morozovalekseywot) [reply vkcs/kubernetes/shared.go]: done

**Pull request merged**